src/bufr/BufrParser/Query/Query.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * (C) Copyright 2022 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <array>
+#include <unordered_map>
+
+#include "QuerySet.h"
+#include "ResultSet.h"
+#include "DataProvider.h"
+
+namespace Ingester {
+namespace bufr {
+    namespace __details
+    {
+        /// \brief The information or Meta data for a BUFR field whose data we wish to capture when
+        /// we execute a query.
+        /// \note Will be refactored to use the SubsetTable object.
+        struct Target {
+            std::string name;
+            std::string queryStr;
+            bool isString;
+            std::vector<int> seqPath;
+            std::vector<int> nodeIds;
+            std::vector<std::string> dimPaths;
+            std::vector<int> exportDimIdxs;
```

Comment:
I assume this is just the index of the dimenision if there are more than 1 dimension?

Reply:
exportDimIdxs keeps track of which dimensions count. This is because "binary" repeats don't count (so they are excluded from this list. I should figure out a cleaner way to do this...